### ISSUE TITLE
2.0 legacy view dynamic filter form is broken

### ISSUE THREAD

**Issue by gsnider2195** [nautobot/nautobot#3858]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0.0a2
* Python version: 3.10.11
* Database platform, version: MySQL
* Middleware(s):

### Steps to Reproduce
1. Go to https://next.demo.nautobot.com/dcim/devices/
2. Open up the filter form and click on the advanced tab
3. Select any filter field and try to select a lookup type

### Expected Behavior

The lookup type dropdown populates with valid options

### Observed Behavior

Nothing loads in the dropdown and the browser generates a JQuery error:

```
jquery-3.6.0.min.js:2 jQuery.Deferred exception: Cannot read properties of undefined (reading 'includes') TypeError: Cannot read properties of undefined (reading 'includes')
    at https://next.demo.nautobot.com/static/js/forms.js?v2.0.0a3:293:40
    at Array.reduce (<anonymous>)
    at n.processResults (https://next.demo.nautobot.com/static/js/forms.js?v2.0.0a3:263:39)
    at Object.<anonymous> (https://next.demo.nautobot.com/static/select2-4.0.13/select2.min.js:2:38054)
    at e (https://next.demo.nautobot.com/static/jquery/jquery-3.6.0.min.js:2:30038)
    at t (https://next.demo.nautobot.com/static/jquery/jquery-3.6.0.min.js:2:30340) undefined
```

**Comment by bryanculver** [comment]
Culprit: https://github.com/nautobot/nautobot/compare/develop...next#diff-8906c95050cdc47ee55226257d68c10271005536e061de29fc6a256d4bbfe102R290-R295

**Comment by HanlinMiao** [comment]
Closed by #3634
